### PR TITLE
[Enhancement](planner)add switch to determine if some unfixed functions will be folded on fe.

### PR DESCRIPTION
**Pull request by sohardforaname**
## Proposed changes

Issue Number: close #xxx

add switch to determine if below functions could be folded on fe.

- now()
- current_date()
- current_time()
- unix_timestamp()
- utc_timestamp()
- uuid()
- rand()

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run fe ut

**Comment by sohardforaname** [comment]
run p0

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ExpressionFunctions.java [hunk 1]
```diff
@@ -67,6 +77,12 @@ public Expr evalExpr(Expr constExpr) {
                 || constExpr instanceof FunctionCallExpr
                 || constExpr instanceof TimestampArithmeticExpr) {
             Function fn = constExpr.getFn();
+            if (ConnectContext.get() != null
+                    && ConnectContext.get().getSessionVariable() != null
+                    && !ConnectContext.get().getSessionVariable().isEnableFoldUnfixedFn()
+                    && unfixedFn.contains(fn.getFunctionName().getFunction())) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ExpressionFunctions.java:83]:
```suggestion
            if ((ConnectContext.get() == null
                    || ConnectContext.get().getSessionVariable() == null
                    || ConnectContext.get().getSessionVariable().isEnableFoldUnfixedFn())
                    && unfixedFn.contains(fn.getFunctionName().getFunction())) {
```

File: fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java [hunk 2]
```diff
@@ -239,6 +239,8 @@ public class SessionVariable implements Serializable, Writable {
 
     public static final String ENABLE_DPHYP_TRACE = "enable_dphyp_trace";
 
+    public static final String ENABLE_FOLD_UNFIXED_FN = "enable_fold_unfixed_fn";
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/SessionVariable.java:242]: UNFIXED --> Nondeterministic

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Review by morrySnow — APPROVED** [review]

**Comment by sohardforaname** [comment]
run buildall

**Comment by sohardforaname** [comment]
run p0

**Comment by sohardforaname** [comment]
run p0

**Pull request merged**